src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,331 @@
+use crate::{
+    commands::toolchain::ExportCommand,
+    constants::{FUEL_TOOLCHAIN_TOML_FILE, VALID_CHANNEL_STR},
+    path::get_fuel_toolchain_toml,
+    toolchain::{DistToolchainDescription, Toolchain},
+    toolchain_override::{self, OverrideCfg, ToolchainCfg, ToolchainOverride},
+    util::version::exec_version,
+};
+use anyhow::{bail, Result};
+use component::{self, Components};
+use forc_tracing::println_warning;
+use semver::Version;
+use std::{collections::HashMap, path::PathBuf};
+use std::{fs, io::BufRead, str::FromStr};
+use tracing::info;
+
+pub fn export(command: ExportCommand, mut reader: impl BufRead) -> Result<()> {
+    let ExportCommand {
+        name,
+        channel,
+        force,
+    } = command;
+    let mut toolchain_info_path = PathBuf::from("./").join(FUEL_TOOLCHAIN_TOML_FILE);
+    if let Some(path) = get_fuel_toolchain_toml() {
+        toolchain_info_path = path;
+        if force {
+            println_warning(&format!(
+                "Because the `--force` argument was supplied, this file will be overwritten: {}",
+                &toolchain_info_path.display(),
+            ));
+            if fs::remove_file(&toolchain_info_path).is_err() {
+                bail!("failed to remove file {}", &toolchain_info_path.display());
+            }
+        } else {
+            println_warning(&format!(
+                "There is an existing toolchain override file at {}. \
+                Do you wish to replace it with a new one? (y/N) ",
+                &toolchain_info_path.display(),
+            ));
+            let mut need_replace = String::new();
+            if reader.read_line(&mut need_replace).is_err() {
+                bail!("failed to read user input");
+            }
+            if need_replace.trim() == "y" {
+                if fs::remove_file(&toolchain_info_path).is_err() {
+                    bail!("failed to remove file {}", &toolchain_info_path.display());
+                };
+            } else {
+                bail!(
+                    "Failed to export toolchain \
+                    because a toolchain override file already exists at {}.",
+                    &toolchain_info_path.display(),
+                );
```

Comment:
I think a more precise message would be to say that we couldn't remove it because the user has declined the replacement rather than it existed

Reply:
@sdankel What do you think of this, before [suggestion](https://github.com/FuelLabs/fuelup/pull/526#discussion_r1424644440)

Reply:
@Braqzen do you have a suggestion for what the error message should be? It's hard to say that without being too wordy. Maybe something like "Canceled toolchain export." would be enough.

Reply:
> Cancelled toolchain export

Is good. It simply states that the action has been discontinued so once the prompt pops up for the user they know that there has been no error. The operation has ended.